tests/test_client.py

```diff
@@ -27,268 +25,338 @@ def get_doc(
     return doc
 
 
-def test_index_list(swg_index_client):
-    r = swg_index_client.list_entries()
-    assert r.records == []
+def test_index_list(client):
+    res = client.get("/index/")
+    rec = res.json
+    assert rec["records"] == []
 
 
-def test_index_list_with_params(swg_index_client):
-    data = get_doc()
-    data["urls"] = ["s3://endpointurl/bucket_2/key_2", "s3://anotherurl/bucket_2/key_2"]
-    data["urls_metadata"] = {
+def test_index_list_with_params(client, user):
+    data1 = get_doc()
+    data1["urls"] = [
+        "s3://endpointurl/bucket_2/key_2",
+        "s3://anotherurl/bucket_2/key_2",
+    ]
+    data1["urls_metadata"] = {
         "s3://endpointurl/bucket_2/key_2": {"state": "error", "other": "xxx"},
         "s3://anotherurl/bucket_2/key_2": {"state": "error", "other": "xxx"},
     }
-    r_1 = swg_index_client.add_entry(data)
+    res_1 = client.post("/index/", json=data1, headers=user)
+    rec_1 = res_1.json
 
     data2 = get_doc()
     data2["metadata"] = {"project_id": "other-project", "state": "abc", "other": "xxx"}
     data2["urls"] = ["s3://endpointurl/bucket/key_2", "s3://anotherurl/bucket/key_2"]
     data2["urls_metadata"] = {
         "s3://endpointurl/bucket/key_2": {"state": "error", "other": "xxx"}
     }
-    r_2 = swg_index_client.add_entry(data2)
-    r = swg_index_client.list_entries(metadata="project_id:bpa-UChicago")
-    ids = [record.did for record in r.records]
-    assert r_1.did in ids
+    res_2 = client.post("/index/", json=data2, headers=user)
+    rec_2 = res_2.json
 
-    r = swg_index_client.list_entries(metadata="project_id:other-project")
-    ids = [record.did for record in r.records]
-    assert r_2.did in ids
+    data1_by_md = client.get("/index/?metadata=project_id:bpa-UChicago")
+    data1_list = data1_by_md.json
+    ids = [record["did"] for record in data1_list["records"]]
+    assert rec_1["did"] in ids
 
-    r = swg_index_client.list_entries(hash="md5:8b9942cf415384b27cadf1f4d2d682e5")
-    ids = [record.did for record in r.records]
-    assert r_1.did in ids
-    assert r_2.did in ids
+    data2_by_md = client.get("/index/?metadata=project_id:other-project")
+    data2_list = data2_by_md.json
+    ids = [record["did"] for record in data2_list["records"]]
+    assert rec_2["did"] in ids
 
-    r = swg_index_client.list_entries(ids=",".join(ids))
+    data_by_hash = client.get("/index/?hash=md5:8b9942cf415384b27cadf1f4d2d682e5")
+    data_list_all = data_by_hash.json
+    ids = [record["did"] for record in data_list_all["records"]]
+    assert rec_1["did"] in ids
+    assert rec_2["did"] in ids
 
-    ids = [record.did for record in r.records]
-    assert r_1.did in ids
-    assert r_2.did in ids
+    idslist = ",".join(ids)
+    data_by_ids = client.get("/index/?ids=" + idslist)
```

Comment:
codacy is complaining about this change, saying: 
"Prohibits string format operations that have more arguments than conversion specifiers declared."

Reply:
my bad, it should be `"/index/?ids={}".format(",".join(ids))`

it worries me that this test is passing even with this mistake. it's because using `?ids=` without a value just returns all the records. could you fix this test so that the list of IDs we're querying is only a subset of all the existing IDs?

Reply:
To do this I queried for only one of the two posted ids